astropy/stats/__init__.py

```diff
@@ -11,35 +11,12 @@
 
 """
 
-from . import funcs
 from .funcs import *  # noqa
-from . import biweight
 from .biweight import *  # noqa
-from . import sigma_clipping
 from .sigma_clipping import *  # noqa
-from . import jackknife
 from .jackknife import *  # noqa
-from . import circstats
 from .circstats import *  # noqa
-from . import bayesian_blocks as _bb
 from .bayesian_blocks import *  # noqa
-from . import histogram as _hist
 from .histogram import *  # noqa
-from . import info_theory
 from .info_theory import *  # noqa
-from . import spatial
 from .spatial import *  # noqa
-from .lombscargle import *  # noqa
-from .bls import *  # noqa
-
-# This is to avoid importing deprecated modules in subpackage star import
-__all__ = []
```

Comment:
This way of populating the namespace is actually quite nice, so I would keep it anyway as opposed to rely on `*` imports.

Reply:
If we are not removing the deprecated modules then this file can't be changed either.